deltatech_product_update/wizard/product_update.py

```diff
@@ -0,0 +1,47 @@
+import base64
+
+from odoo import api, fields, models
+
+try:
+    import xlrd
+
+    try:
+        from xlrd import xlsx
+
+    except ImportError:
+        xlsx = None
+except ImportError:
+    xlrd = xlsx = None
+
+
+class ProductImportFile(models.TransientModel):
+    _name = "product.import.file"
+    _description = "Import product file"
+    data_file = fields.Binary(string="File", required=True)
+    vendor_id = fields.Many2one("res.partner")
+
+    @api.depends("vendor_id")
+    def do_import(self):
+        decoded_data = base64.b64decode(self.data_file)
+        book = xlrd.open_workbook(file_contents=decoded_data)
+        sheet = book.sheet_by_index(0)
+        table_values = []
+        for row in list(map(sheet.row, range(sheet.nrows))):
+            values = []
+            for cell in row:
+                if cell.ctype is xlrd.XL_CELL_NUMBER:
+                    is_float = cell.value % 1 != 0.0
+                    values.append(str(cell.value) if is_float else str(int(cell.value)))
+                else:
+                    values.append(cell.value)
+            table_values.append(values)
+
+        for key in table_values:
+            product = self.env["product.supplierinfo"].search(
+                [("name", "=", self.vendor_id.name), ("product_tmpl_id.default_code", "=", key[0])]
```

Comment:
in product.supplierinfo campul name este un id de partener deci ("name", "=", self.vendor_id.id)
codul furnizorului este stocat in product_code vezi https://github.com/odoo/odoo/blob/14.0/addons/product/models/product.py#L743
("product_code", "=", key[0])

Reply:
Nu cred ca  merge cu self.vendor_id.name

Reply:
Am verificat si mie imi merge, wizard-ul functrioneaza fara erori,
![Screenshot from 2021-08-08 14-14-48](https://user-images.githubusercontent.com/56277656/128630210-20d40e35-6fbf-4f64-b5bf-5f1e660db769.png)
Dar pentru siguranta ar trebuie verificat de altcineva